mysys/my_getopt.c

```diff
@@ -1586,6 +1588,21 @@ void my_print_help(const struct my_option *options)
 	col+= (optp->arg_type == OPT_ARG) ? 5 : 3;
       }
     }
+    if (optp->deprecation_substitute != NULL)
+    {
+      if (strcmp(optp->deprecation_substitute, OPT_DEPRECATED_NO_REPLACEMENT) == 0)
+        col= print_comment("(DEPRECATED)",
+                           col, name_space, comment_space);
+      else
+      {
+        col= print_comment("(DEPRECATED and replaced by '",
+                           col, name_space, comment_space);
+        col= print_comment(optp->deprecation_substitute,
+                           col, name_space, comment_space);
+        col= print_comment("')",
+                           col, name_space, comment_space);
+      }
+    }
```

Comment:
may be closer to what the server was printed before? And why are you SHOUTING?
`(deprecated)` and `(deprecated, use %s instead)` would be better

Reply:
I don't see  a warning when a deprecated option is used. Like in `mysqld --debug`

Reply:
> I don't see a warning when a deprecated option is used. Like in mysqld --debug

I didn't think about that part, the current PR only add the logic to print the deprecation message for the `mysqld --help --verbose` output and when the variable is set via `SET var_name`. But I think we should add that too, if you agree, for the next revision I can add the logic for printing the warning at the bottom of:
https://github.com/MariaDB/server/blob/c37ebaf6c224e7ba571653eecd48a080d4c1a061/sql/mysqld.cc#L7967